### PR TITLE
feature/provider-refresh-and-task-check-actions

### PR DESCRIPTION
**Pull request by jschoewe**
Added actions to refresh providers and check the status of a task. Also fixed circle ci python 3 errors.

### CODE REVIEW

File: actions/lib/providers.py [hunk 1]
```diff
@@ -42,3 +42,32 @@ def credentials_test(self, client, kwargs_dict):
...
+            for prov in providers:
+                if str(prov['id']) == kwargs_dict['provider_id']:
+                    resources.append(prov)
+                    break
+        else:
+            resources = providers
+
+        endpoint = "https://" + kwargs_dict['server'] + "/api/providers"
+
+        result = client.post(url=endpoint, action="refresh", resources=resources)
```
Comment (nmaludy) [inline actions/lib/providers.py]: there should be no reason we need to construct this manually, the client should be able to have something along these lines... maybe like `client.collections.providers.refresh(resources)` or something... would need to dig in and play around, but it should be as simple as that without building the URL ourselves

File: actions/lib/task_check_complete.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    def task_check_complete(self, client, kwargs_dict):
+        """Check if a given task is complete and fail if it isn't
+        :param client: connection from the base_action
+        :param kwargs_dict:
+        """
+        id = self._get_arg("task_id", kwargs_dict)
+        task = client.collections.tasks(id)
+
+        if(task.state.lower() != 'finished'):
```
Comment (nmaludy) [inline actions/lib/task_check_complete.py:21]: are there other possible states (maybe errors, etc) that this could be in?

Comment (jschoewe) [reply actions/lib/task_check_complete.py:21]: Errors show up in the "status" field. Possible States are Active, Finished, Initialized, and Queued

File: actions/lib/providers.py [hunk 3]
```diff
@@ -42,3 +42,34 @@ def credentials_test(self, client, kwargs_dict):
...
+                if str(prov['id']) == kwargs_dict['provider_id']:
+                    resources.append(prov)
+                    break
+        else:
+            resources = providers
+
+        result = client.collections.providers.action.refresh(*resources)
+
+        # return the data objects from each of the action results
+        results = []
```
Comment (nmaludy) [inline actions/lib/providers.py]: could probably use this: _data_from_entity_list()
https://github.com/StackStorm-Exchange/stackstorm-manageiq/blob/master/actions/lib/base_action.py#L44-L48

**Pull request merged**